src/WebPage.cpp

```diff
@@ -83,13 +83,15 @@ void WebPage::setFrameProperties(QWebFrame *frame, QUrl &requestedUrl, NetworkRe
   if (frame->requestedUrl() == requestedUrl) {
     int statusCode = reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
     frame->setProperty("statusCode", statusCode);
-    QVariantMap headers;
-    foreach(QNetworkReply::RawHeaderPair header, reply->rawHeaderPairs())
-      headers[header.first] = QString(header.second);
-    frame->setProperty("headers", headers);
-    frame->setProperty("body", reply->data());
-    QVariant contentMimeType = reply->header(QNetworkRequest::ContentTypeHeader);
-    frame->setProperty("contentType", contentMimeType);
+    if (statusCode != 304) {
```

Comment:
In my testing, the frame object is reused. You can see the one test I have for this here: https://github.com/thoughtbot/capybara-webkit/pull/775/files#diff-089b82733ec96b02481c41d93683d617R2130

This area does concern me, as I couldn't find any behavior around the documentation for cached pages and the Qt/WebKit code is fairly dense. I wouldn't be surprised if there were more monsters lurking in this area, but I think we'll need to get people to use it in order to find out what I may have missed.